compiler/src/main/java/io/neow3j/compiler/NeoEvent.java

```diff
@@ -25,17 +29,43 @@ public class NeoEvent {
     private final FieldNode asmVariable;
 
     public NeoEvent(FieldNode asmVariable, ClassNode owner) {
-        Optional<AnnotationNode> annOpt = AsmHelper.getAnnotationNode(asmVariable, DisplayName.class);
-        if (annOpt.isPresent()) {
-            displayName = (String) annOpt.get().values.get(1);
+        Optional<AnnotationNode> displayNameAnnOpt = AsmHelper.getAnnotationNode(asmVariable, DisplayName.class);
+        if (displayNameAnnOpt.isPresent()) {
+            displayName = (String) displayNameAnnOpt.get().values.get(1);
         } else {
             displayName = asmVariable.name;
         }
 
-        AtomicInteger argNr = new AtomicInteger(1);
-        params = extractTypeParametersFromSignature(asmVariable).stream()
-                .map(t -> new Param(PARAM_NAME_PREFIX + argNr.getAndIncrement(), Type.getType(t)))
-                .collect(Collectors.toList());
+        Optional<AnnotationNode> paramNamesAnnOpt = AsmHelper.getAnnotationNode(asmVariable, EventParameterNames.class);
+        if (paramNamesAnnOpt.isPresent()) {
+            List<Object> eventParamNameValues = paramNamesAnnOpt.get().values;
+            assert eventParamNameValues.size() >= 2 : "The EventParameterNames annotation should never have less than" +
+                    " 2 entries.";
+            ArrayList paramNames = (ArrayList) eventParamNameValues.get(1);
+
+            AtomicInteger paramNr = new AtomicInteger(0);
+            params = extractTypeParametersFromSignature(asmVariable).stream()
+                    .map(t -> {
+                        // Check if there are enough parameter names provided for the event.
+                        if (paramNames.size() < paramNr.get() + 1) {
+                            throw new CompilerException(format("Not enough parameter names provided for event %s.",
+                                    displayName));
+                        }
+                        return new Param((String) paramNames.get(paramNr.getAndIncrement()), Type.getType(t));
+                    })
+                    .collect(Collectors.toList());
+            // After the lambda, all parameter names should have been used, i.e., the atomic int paramNr should now be
+            // equal to the number of parameters.
+            if (paramNames.size() != paramNr.get()) {
+                throw new CompilerException(format("Too many parameter names provided for event %s.", displayName));
```

Comment:
This message is maybe misleading... AFAIK, this if statement will also get "too few" param names, right?

Reply:
Without context, you're right, but if it's too few, an exception will be thrown some lines above in the lamdba expression.